plaso/preprocessors/mediator.py

```diff
@@ -113,11 +113,12 @@ def AddUserAccount(self, user_account):
       KeyError: if the user account already exists.
     """
     if not user_account.username:
-      logger.debug(f'adding user account: {user_account.identifier:s}')
+      logger.debug('adding user account: {0!s}'.format(user_account.username))
     else:
-      logger.debug(
-          f'adding user account: {user_account.username:s} '
-          f'({user_account.identifier:s})')
+      logger.debug('adding user account: {0!s} ({1!s})'.format(
```

Comment:
why is identifier not set or not a string in your case?

Reply:
In our case the username value was "None" when in WindowsUserAccountsPlugin / _ParseKey
Looking more closely at the code, I think it's probably because in this section, if you fall in the first condition:
```
if not registry_value:
  username = 'N/A'
else:
  profile_path = registry_value.GetDataAsObject()
  username = self._GetUsernameFromProfilePath(profile_path)

  user_account.user_directory = profile_path or None
  user_account.username = username or None
```

Then neither `user_account. user_directory ` nor `user_account.username`  is set.
Which then fail in  `mediator.AddUserAccount(user_account)`, but not with a KeyError

Reply:
can you provide the edge case data why identifier is not set?

Reply:
I'm asking about identifier not username that is the one where you are using `!s`